app/controllers/personal_access_tokens_controller.rb

```diff
@@ -0,0 +1,38 @@
+class PersonalAccessTokensController < ApplicationController
+  include Foreman::Controller::Parameters::PersonalAccessToken
+  include Foreman::Controller::SshKeysCommon
+  include Foreman::Controller::ActionPermissionDsl
+
+  before_action :find_resource, :only => [:revoke]
+
+  define_action_permission 'revoke', :revoke
+
+  def new
+    @personal_access_token = PersonalAccessToken.new
+  end
+
+  def create
+    @personal_access_token = PersonalAccessToken.new(personal_access_token_params.merge(:user => @user))
+    token_value = @personal_access_token.generate_token
+    if @personal_access_token.save
+      flash[:personal_access_token] = token_value
```

Comment:
so why not simply pass that as a @variable to the view? why do you need to use flash?